scripts/split_repo_migration/git-migrate-history.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+if [ $# -lt 4 ]
+then
+  echo "Usage: $0 <source-repo> <target-repo> <source-path> <target-path> [folders,to,skip] [files,to,keep] [branch-name]"
+  exit 1
+fi
+
+# source GitHub repository. format: <owner>/<repo>
+SOURCE_REPO=$1
+
+# destination GitHub repository. format: <owner>/<repo>
+TARGET_REPO=$2
+
+# path in the source repo to copy code from. Defaults to the root directory
+SOURCE_PATH=$3
+
+# path in the target repo to put the copied code
+TARGET_PATH=$4
+
+# comma-separated list of files/folders to skip
+IGNORE_FOLDERS=$5
+# keep these specific files that would otherwise be deleted by IGNORE_FOLDERS
+KEEP_FILES=$6
+
+# override the HEAD branch name for the migration PR
+BRANCH=$7
+
+# Path for update script for migration postprocessing.
+UPDATE_SCRIPT=$8
```

Comment:
IIUC, this could be an arbitrary path in the file system, so it could have spaces we should enclose it in quotes: `"$8"`. In most cases, it's advisable to quote shell variable references for safety.

Reply:
Removed update script.